mPowerSDK/TasksAndSteps/MedicationTracker/APHMedicationTrackerTask.m

```diff
@@ -0,0 +1,717 @@
+            [items addObject:item];
+        }
+    }
+    
+    formStep.formItems = items;
+    
+    return (items.count > 0);
+}
+
+- (BOOL)shouldUpdateAndIncludeMomentInDayStep:(ORKStep *)step {
+    
+    ORKFormStep *formStep = (ORKFormStep *)step;
+    if (![formStep isKindOfClass:[ORKFormStep class]]) {
+        NSAssert(NO, @"Medication frequency step is not of expected class");
+        return NO;
+    }
+    
+    if ((self.subTask == nil) || !self.dataStore.shouldIncludeMomentInDayStep) {
+        return NO;
+    }
+
+    // Get the medication list
+    NSArray *medList = self.dataStore.trackedMedications;
+    
+    // Build list of meds
+    NSString *itemTextFormat = NSLocalizedStringWithDefaultValue(@"APH_MOMENT_IN_DAY_QUESTION", nil, APHLocaleBundle(), @"When was the last time you took your %@?", @"Prompt for timing of medication in pre-activity medication timing survey where %@ is a list of medications (For example, 'Levodopa or Rytary')");
+    NSString *orWord = NSLocalizedStringWithDefaultValue(@"APH_OR_FORMAT", nil, APHLocaleBundle(), @"or", @"Format of a list with two items using the OR key word.");
+    NSString *listDelimiter = NSLocalizedStringWithDefaultValue(@"APH_LIST_FORMAT_DELIMITER", nil, APHLocaleBundle(), @",", @"Delimiter for a list of more than 3 items. (For example, 'Levodopa, Simet or Rytary')");
+    
+    NSMutableString *listText = [NSMutableString new];
+
+    [medList enumerateObjectsUsingBlock:^(NSString * _Nonnull obj, NSUInteger idx, BOOL * _Nonnull stop __unused) {
+        if (medList.count > 1) {
+            if (idx+1 == medList.count) {
+                [listText appendFormat:@" %@ ", orWord];
+            }
+            else if (idx != 0) {
+                [listText appendFormat:@"%@ ", listDelimiter];
+            }
+        }
+        [listText appendString:obj];
+    }];
+    NSString *itemText = [NSString stringWithFormat:itemTextFormat, listText];
+
+    // If the steps include the medication introduction, do not include the explanation text here
+    if ([self previousStepWithIdentifier:APHMedicationTrackerIntroductionStepIdentifier] != nil) {
+        step.text = itemText;
+        itemText = nil;
+    }
+    
+    ORKAnswerFormat  *format = [ORKTextChoiceAnswerFormat
+                                choiceAnswerFormatWithStyle:ORKChoiceAnswerStyleSingleChoice
+                                textChoices:[self momentInDayChoices]];
+    
+    ORKFormItem  *item = [[ORKFormItem alloc] initWithIdentifier:APHMedicationTrackerMomentInDayFormItemIdentifier
+                                                            text:itemText
+                                                    answerFormat:format];
+    [formStep setFormItems:@[item]];
+    
+    return YES;
+}
+
+- (NSArray <ORKTextChoice *> *) momentInDayChoices
+{
+    NSString *formatMinutes = NSLocalizedStringWithDefaultValue(@"APH_MINUTES_RANGE_FORMAT", nil, APHLocaleBundle(),
```

Comment:
Seems like it would be simpler, and definitely more localization-robust, to just add the four interval choices as localized string literals rather than composing them like this. Or am I missing something?